common/src/main/java/io/novafoundation/nova/common/utils/stateMachine/list/PaginatedListStateMachine.kt

```diff
@@ -0,0 +1,51 @@
+package io.novafoundation.nova.common.utils.stateMachine.list
+
+import io.novafoundation.nova.common.data.model.DataPage
+import io.novafoundation.nova.common.data.model.PageOffset
+import io.novafoundation.nova.common.utils.stateMachine.StateMachine
+import io.novafoundation.nova.common.utils.stateMachine.list.states.EmptyProgressState
+import io.novafoundation.nova.common.utils.stateMachine.list.states.PaginatedListState
+import kotlinx.coroutines.CoroutineScope
+
+object PaginatedListStateMachine {
+
+    sealed class Event<out T> {
+
+        companion object {
+
+            fun TriggerInitialLoading(): Event<Nothing> {
```

Comment:
In Nova Wallet, we have `bootstrap` method in the `state` that can be used to execute some initial logic.
We can use it instead of using this approarch